hack/deployer/config/plans.yml

```diff
@@ -164,10 +164,10 @@ plans:
   clusterName: kind-ci
   clientVersion: 0.11.1
```

Comment:
Also bump the kind version to [v0.14.0](https://github.com/kubernetes-sigs/kind/releases/tag/v0.14.0) (duplicated in the `kind-dev` profile)?